pandas/core/generic.py

```diff
@@ -11222,6 +11222,10 @@ def logical_func(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs
                 raise NotImplementedError(
                     "Option bool_only is not implemented with option level."
                 )
+            if isinstance(self, ABCSeries) and self.dtype.name == "boolean":
```

Comment:
we would want to do this more generically. can you do this deepper in _agg_by_level; ultimately the EA itself should be returning a coerced result.

Reply:
Done!